node/node.go

```diff
@@ -421,3 +421,14 @@ func (nn nexNode) SetLameDuck(ctx context.Context) {
 		nn.interrupt <- os.Interrupt
 	}()
 }
+
+func (nn nexNode) IsTargetNode(inId string) (bool, error) {
+	pub, err := nn.publicKey.PublicKey()
+	if err != nil {
+		return false, err
+	}
+	if inId == pub || inId == nn.auctionBidId {
```

Comment:
Using a single auction bid ID here is problematic. For example, if two different clients perform an auction, then when the second one issues a deploy request, the bid ID will be gone because the node has only a single bid ID.

In short, _every_ reply to an auction should be a unique bid ID. The bid IDs can be stored in a map between the auction request and the positive bid. Maybe this means each auction needs its own nuid as well, so we can create a `map[bidId]auctionID`.

The node can expire these bid IDs every 30s or so. When a deploy/run request comes into the node for a given auction ID, if there's no corresponding auction ID, then the deploy request is rejected.

Reply:
I literally realized this as I was getting in bed last night 🤣